src/main/java/hudson/plugins/promoted_builds/integrations/jobdsl/JobDslPromotionProcess.java

```diff
@@ -32,7 +33,7 @@
     /**
      * {@link PromotionCondition}s. All have to be met for a build to be promoted.
      */
-    private List<PromotionCondition> conditions = new ArrayList<PromotionCondition>();
+    private List<Node> conditions = new ArrayList<Node>();
```

Comment:
From what I see it's going to cause conflicts when the data gets deserialized by XStream. 🐛